static/app/views/issueDetails/traceTimeline/traceIssue.tsx

```diff
@@ -68,18 +71,44 @@ export function TraceIssueEvent({event}: TraceIssueEventProps) {
   );
 }
 
+// This function tries to imitate what getTitle() from utils.events does.
+// In that module, the data comes from the issues endpoint while in here
+// we grab the data from the events endpoint. A larger effort is
+// required in order to use that function directly since the data between
+// the two endpoint is slightly different.
+// For instance, the events endpoint could include a _metadata dict with
+// the title, subtitle and message.
+// We could also make another call to the issues endpoint  to fetch the metadata,
+// however, we currently don't support it and it is extremely slow
 function getTitleSubtitleMessage(event: TimelineEvent) {
   let title;
-  // XXX: This is not fully correct but it will make this first PR easier to review
-  const subtitle = event.transaction;
-  let message = event.message;
-  if (event['event.type'] === 'error') {
-    title = event.title.split(':')[0];
-  } else {
-    title = event.title;
-    // It is suspected that this value is calculated somewhere in Relay
-    // and we deconstruct it here to match what the Issue details page shows
-    message = event.message.replace(event.transaction, '').replace(title, '');
+  let subtitle = event.culprit || '';
+  let message;
+  try {
+    title = event.title.trimEnd();
+    if (event['event.type'] === 'error') {
+      if (title[title.length - 1] !== ':') {
+        title = event.title.split(':')[0];
+      }
+      // https://github.com/getsentry/sentry/blob/f08644a004f9980d48f93dec2d7cfc9eeecd9a9e/static/app/utils/events.tsx#L48-L50
+      // It uses metadata.value which could differ depending on what error.value is used in the event manager
+      message = event['error.value'][event['error.value'].length - 1];
+    } else if (event['event.type'] === 'default') {
+      // https://github.com/getsentry/sentry/blob/f08644a004f9980d48f93dec2d7cfc9eeecd9a9e/static/app/utils/events.tsx#L179-L184
+      subtitle = '';
+      // https://github.com/getsentry/sentry/blob/f08644a004f9980d48f93dec2d7cfc9eeecd9a9e/static/app/utils/events.tsx#L59-L60
+      message = event.culprit || '';
+    } else {
+      const issuePlatformEvent = event as TimelineIssuePlatformEvent;
+      // It is suspected that this value is calculated somewhere in Relay
+      // and we deconstruct it here to match what the Issue details page shows
+      message = issuePlatformEvent.message
```

Comment:
This is as it was:
<img width="499" alt="image" src="https://github.com/getsentry/sentry/assets/44410/a83e4499-b36e-45cf-b9f0-8710cc123902">